example-app/app/reducers/index.ts

```diff
@@ -0,0 +1,161 @@
+import {
+  ActionReducerMap,
+  createSelector,
+  createFeatureSelector,
+  compose,
+  ActionReducer,
+  combineReducers,
+  Action,
+  ActionReducerFactory,
+} from '@ngrx/store';
+import * as fromRouter from '@ngrx/router-store';
+import { environment } from '../../environments/environment';
+
+
+/**
+ * Every reducer module's default export is the reducer function itself. In
+ * addition, each module should export a type or interface that describes
+ * the state of the reducer plus any selector functions. The `* as`
+ * notation packages up all of the exports into a single object.
+ */
+import * as fromSearch from 'app/books/reducers/search';
+import * as fromBooks from 'app/books/reducers/books';
+import * as fromCollection from 'app/books/reducers/collection';
+import * as fromLayout from 'app/layout/reducers/layout';
+
+
+/**
+ * As mentioned, we treat each reducer like a table in a database. This means
+ * our top level state interface is just a map of keys to inner state types.
+ */
+export interface State {
+  search: fromSearch.State;
+  books: fromBooks.State;
+  collection: fromCollection.State;
+  layout: fromLayout.State;
+}
+
+
+/**
+ * Because metareducers take a reducer function and return a new reducer,
+ * we can use our compose helper to chain them together. Here we are
+ * using combineReducers to make our top level reducer, and then
+ * wrapping that in storeLogger. Remember that compose applies
+ * the result from right to left.
+ */
+export const reducers: ActionReducerMap<State> = {
+  search: fromSearch.reducer,
+  books: fromBooks.reducer,
+  collection: fromCollection.reducer,
+  layout: fromLayout.reducer,
+};
+
+
+// console.log all actions
+function logger(reducer: ActionReducer<State>) {
+  return function(state: State, action: any) {
+    console.log('state', state);
+    console.log('action', action);
+
+    return reducer(state, action);
+  }
+}
+
+/**
+ * The compose function is one of our most handy tools. In basic terms, you give
```

Comment:
I don't think many of these comments apply anymore. Maybe add a comment about what `createSelector` does?